lib/models/assignments.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:mobile_app/models/grade.dart';
+import 'package:mobile_app/models/links.dart';
+import 'package:mobile_app/models/projects.dart';
+
+class Assignments {
+  Assignments({
+    this.data,
+    this.links,
+  });
+
+  List<Assignment> data;
+  Links links;
+
+  factory Assignments.fromJson(Map<String, dynamic> json) => Assignments(
+        data: List<Assignment>.from(
+          json['data'].map((x) => Assignment.fromJson(x)),
+        ),
+        links: Links.fromJson(json['links']),
+      );
+}
+
+class Assignment {
+  Assignment({
+    this.id,
+    this.type,
+    this.attributes,
+    this.projects,
+    this.grades,
+  });
+
+  String id;
+  String type;
+  AssignmentAttributes attributes;
+  List<Project> projects;
+  List<Grade> grades;
+
+  factory Assignment.fromJson(Map<String, dynamic> json) => Assignment(
+        id: json['id'] ?? json['data']['id'],
+        type: json['type'] ?? json['data']['type'],
+        attributes: AssignmentAttributes.fromJson(
```

Comment:
https://pub.dev/packages/json_api use libraries for serialization

Reply:
Will see to it, a bit of research would be required to get to a conclusion if this could a good fit.. @tachyons